lib/livebook/teams/connection.ex

```diff
@@ -73,9 +70,12 @@ defmodule Livebook.Teams.Connection do
         Process.send_after(self(), {:loop_ping, data.ref}, @loop_ping_delay)
         {:keep_state, %{data | http_conn: conn, websocket: websocket}}
 
-      {:error, conn, websocket, _reason} ->
-        Logger.warning("Teams WebSocket connection - ping error")
-        {:keep_state, %{data | http_conn: conn, websocket: websocket}}
+      {:error, conn, websocket, reason} ->
+        data = %{data | http_conn: conn, websocket: websocket}
+        send(data.listener, {:connection_error, reason})
+        Logger.warning("Teams WebSocket connection - ping error: #{inspect(reason)}")
+        ensure_closed(data)
+        {:keep_state, data, {:next_event, :internal, :connect}}
```

Comment:
Currently, when ping fails, we only log the message. We neither schedule another ping, nor reconnect, which doesn't sound right. I think it makes sense to treat ping failure in the same way as regular send failure and just reconnect.

Reply:
FWIW with this logic we would automatically recover from the bug fixed in this PR, since we either ping successfully or reconnect, but never get stuck.